### PR TITLE
Add soundness and completeness to glossary

### PR DESCRIPTION
**Pull request by mark-i-m**
See #48 

cc @phansch

### CODE REVIEW

File: src/glossary.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Term                    | Meaning
 ------------------------|--------
 AST                     |  the abstract syntax tree produced by the syntax crate; reflects user syntax very closely.
 codegen unit            |  when we produce LLVM IR, we group the Rust code into a number of codegen units. Each of these units is processed by LLVM independently from one another, enabling parallelism. They are also the unit of incremental re-use.
+completeness            |  soundness is a technical term in type theory. Completeness means that every type-safe program also type-checks. Having both soundness and completeness is very hard, and usually soundness is more important. (see "soundness").
```
Comment (MaloJaffre) [inline src/glossary.md]: typo: soundness is ...

File: src/glossary.md [hunk 2]
```diff
@@ -25,6 +26,7 @@ provider                |  the function that executes a query ([see more](query.
 query                   |  perhaps some sub-computation during compilation ([see more](query.html))
 sess                    |  the compiler session, which stores global data used throughout compilation
 side tables             |  because the AST and HIR are immutable once created, we often carry extra information about them in the form of hashtables, indexed by the id of a particular node.
+soundness               |  soundness is a technical term in type theory. Roughly, if a type system is sound, then if a program type-checks, it is type-safe; i.e. I can never (in safe rust) force a value into a variable of the wrong type. (see "completeness").
```
Comment (MaloJaffre) [inline src/glossary.md]: typo: I can never ...

Comment (mark-i-m) [reply src/glossary.md]: Sorry, what is the typo here?

Comment (MaloJaffre) [reply src/glossary.md]: Oops sorry, I wrongly read this sentence.

**Pull request merged**